packages/editor-ui/src/components/mixins/externalHooks.ts

```diff
@@ -0,0 +1,38 @@
+import Vue from "vue";
+import { IExternalHooks } from "@/Interface";
+import { IDataObject } from "n8n-workflow";
+
+export async function runExternalHook(
+	eventName: string,
+	// tslint:disable-next-line: no-any
+	state: any,
+	metadata?: IDataObject,
+) {
+	// @ts-ignore
+	if (!window.externalHooks) {
+		return;
+	}
+
+	const [resource, operator] = eventName.split(".");
+
+	// @ts-ignore
+	if (window.externalHooks[resource] && window.externalHooks[resource][operator]) {
+		// @ts-ignore
+		const hookMethods = window.externalHooks[resource][operator];
+		for (const hookmethod of hookMethods) {
+			hookmethod(state, metadata);
+		}
+	}
+}
+
+export const externalHooks = Vue.extend({
+	methods: {
+		externalHooks(): IExternalHooks {
+			return {
+				onExternalHookEvent: (eventName: string, metadata?: IDataObject): void => {
+					runExternalHook(eventName, this.$store.state, metadata);
```

Comment:
Should have "await".

Reply:
Is there a reason why we provide the store.state? Should it not rather be the store itself to be able to use getters and commit? Or is it not simply better to use `.call(this, ...` to stay flexible?